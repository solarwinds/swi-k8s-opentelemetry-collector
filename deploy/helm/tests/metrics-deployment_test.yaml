# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for metrics-deployment
templates:
  - metrics-deployment.yaml
  - metrics-collector-config-map.yaml
  - common-env-config-map.yaml
  - metrics-collector-env-config-map.yaml
tests:
  - it: Metrics collector spec should match snapshot when using default values
    template: metrics-deployment.yaml
    asserts:
      - matchSnapshot:
          path: spec.template.spec