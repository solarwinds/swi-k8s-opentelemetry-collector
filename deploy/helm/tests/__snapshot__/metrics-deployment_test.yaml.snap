Metrics collector spec should match snapshot when using default values:
  1: |
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
                - key: kubernetes.io/arch
                  operator: In
                  values:
                    - amd64
                    - arm64
                - key: kubernetes.io/os
                  operator: In
                  values:
                    - linux
    containers:
      - command:
          - /swi-otelcol
          - --config=/conf/relay.yaml
        env:
          - name: MY_POD_IP
            valueFrom:
              fieldRef:
                apiVersion: v1
                fieldPath: status.podIP
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-metrics-env-config
        image: solarwinds/swi-opentelemetry-collector:0.8.12
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: 13133
          initialDelaySeconds: 10
        name: swi-opentelemetry-collector
        ports:
          - containerPort: 4317
            name: otlp
            protocol: TCP
          - containerPort: 8888
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: 13133
          initialDelaySeconds: 10
        resources:
          limits:
            memory: 3Gi
          requests:
            memory: 3Gi
        securityContext: {}
        volumeMounts:
          - mountPath: /conf
            name: opentelemetry-collector-configmap
            readOnly: true
    initContainers:
      - command:
          - /bin/grpcurl
          - -expand-headers
          - -proto
          - opentelemetry/proto/collector/logs/v1/logs_service.proto
          - -H
          - 'Authorization: Bearer ${SOLARWINDS_API_TOKEN}'
          - -d
          - '{"resourceLogs":[{"resource":{"attributes":[{"key":"sw.k8s.cluster.uid","value":{"stringValue": "<CLUSTER_UID>"}}]},"scope_logs":{"log_records":[{"severityText":"INFO","body":{"stringValue":"otel-endpoint-check successful"}}]}}]}'
          - <OTEL_ENVOY_ADDRESS>
          - opentelemetry.proto.collector.logs.v1.LogsService.Export
        env:
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
        image: fullstorydev/grpcurl:v1.8.7@sha256:ee1a84e31a5f99af12e0767314c59f05a578c01d28404049a3964d67a15b3580
        imagePullPolicy: IfNotPresent
        name: otel-endpoint-check
        volumeMounts:
          - mountPath: opentelemetry/proto/collector/logs/v1/logs_service.proto
            name: opentelemetry-collector-configmap
            readOnly: true
            subPath: logs_service.proto
          - mountPath: opentelemetry/proto/logs/v1/logs.proto
            name: opentelemetry-collector-configmap
            readOnly: true
            subPath: logs.proto
          - mountPath: opentelemetry/proto/common/v1/common.proto
            name: opentelemetry-collector-configmap
            readOnly: true
            subPath: common.proto
          - mountPath: opentelemetry/proto/resource/v1/resource.proto
            name: opentelemetry-collector-configmap
            readOnly: true
            subPath: resource.proto
    securityContext: {}
    serviceAccountName: RELEASE-NAME-swo-k8s-collector
    terminationGracePeriodSeconds: 600
    volumes:
      - configMap:
          items:
            - key: metrics.config
              path: relay.yaml
            - key: logs_service.proto
              path: logs_service.proto
            - key: logs.proto
              path: logs.proto
            - key: common.proto
              path: common.proto
            - key: resource.proto
              path: resource.proto
          name: RELEASE-NAME-swo-k8s-collector-metrics-config
        name: opentelemetry-collector-configmap
